### ISSUE TITLE
Control Focus Order aka Tab Order

### ISSUE THREAD

**Issue by Goltanju** [fyne-io/fyne#4106]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

This might just be a question if this is already implemented and I just missed the documentation for it. Or maybe it's just a documentation request, again, if already implemented. :)

I'm having a hard time determining how Fyne determines tab/focus order. I'm guessing it's just "the order they were added to the list of stuff to render" but where is that list, can I modify or somehow control its order, etc.?

It seems some layouts change focus order to be more inline with what one would probably want with that layout? Such as the border layout making the "main" object the first focusable even though it's like the fifth object added. etc.

Put this on the back burner as I know you all are finalizing a release right now. Just a request for more info, more docs, or if not implemented yet the plans to do so.

### Is it possible to construct a solution with the existing API?

I don't think so.

### Describe the solution you'd like to see.

I think as it is, having to not think about focus order is just fine for most cases.

But if the layout and the focus order reponsibilities are 100% glued it makes some things seemingly impossible, such as first tab hits top of a border layout, next tab hits main content, etc.

One way I could imagine solving this would be to have a parallel list of focusable objects, in the order desired. If empty, the current behaviour would continue. But otherwise, the parallel list would be used first, then any focusables that weren't in the parallel list, then wrap back around.

**Comment by andydotxyz** [comment]
You're right it is the order added. This is the order of items in the Container, also the draw order.

**Comment by matwachich** [comment]
Discussed here https://github.com/fyne-io/fyne/issues/1515#issuecomment-1299950675
